pkg/manager/member/tikv_group_member_manager.go

```diff
@@ -426,9 +440,18 @@ func getTikVConfigMapForTiKVGroup(tg *v1alpha1.TiKVGroup, tc *v1alpha1.TidbClust
 	if tg.Spec.Config == nil {
 		tg.Spec.Config = &v1alpha1.TiKVConfig{}
 	}
+	version, err := semver.NewVersion(tc.TiKVVersion())
+	if err != nil {
+		return nil, fmt.Errorf("tikvgroup[%s/%s] failed to parse version,err:%v", tg.Namespace, tg.Name, err)
+	}
+	enableAdvertiseStatusAddr := true
+	// For the version less than v4.0.1, don't use advertise-status-addr start arg
+	if version.LessThan(tikvV401Version) {
+		enableAdvertiseStatusAddr = false
+	}
 	scriptModel := &TiKVStartScriptModel{
 		Scheme:                    tc.Scheme(),
-		EnableAdvertiseStatusAddr: true,
+		EnableAdvertiseStatusAddr: enableAdvertiseStatusAddr,
```

Comment:
what if `tc.spec.EnableDynamicConfiguration` is set to `false`? will it cause problems?

Reply:
If tikv version is under v4.0.1, we should not set advertise-status-addr otherwise it will cause start failed.

Reply:
I mean it's different with here

https://github.com/pingcap/tidb-operator/blob/6d9a84c179eb60c23587943eae790327b95051e3/pkg/manager/member/tikv_member_manager.go#L573-L581

If `tc.spec.EnableDynamicConfiguration` is `false`, tikv replicas deployed by `TidbCluster` will not report to PD, but replicas deployed by `TikvGroup` will report to PD.

I'm fine with this if it's your intention. I guess this will not cause problems.

Reply:
TiKV will always be reported to PD in version v4.0.0, `EnableDynamicConfiguration` is used to control whether send `advertise-status-addr` or `advertise-addr` to PD.